src/common/models/time-shift/time-shift.mocha.ts

```diff
@@ -40,4 +44,89 @@ describe("isValidTimeShift", () => {
     expect(isValidTimeShift("P2M"), "two months").to.be.true;
     expect(isValidTimeShift("P5Y"), "five years").to.be.true;
   });
+
+  describe("constrainToFilter", () => {
+
+    const shift = TimeShift.fromJS("P3D");
+
+    describe("Fixed time filter", () => {
+      it("does not touch if shifted period do not overlap with original", () => {
+        const filter = new FixedTimeFilterClause({
+          reference: "time",
+          values: List.of(new DateRange({ start: new Date("2010-01-01"), end: new Date("2010-01-02") }))
+        });
+        expect(shift.constrainToFilter(filter, Timezone.UTC).equals(shift)).to.be.true;
+      });
+
+      it("returns empty time shift if shifted period overlap with original", () => {
+        const filter = new FixedTimeFilterClause({
+          reference: "time",
+          values: List.of(new DateRange({ start: new Date("2010-01-01"), end: new Date("2010-01-04") }))
+        });
+        expect(shift.constrainToFilter(filter, Timezone.UTC).equals(TimeShift.empty())).to.be.true;
+      });
+    });
+
+    describe("Relative time filter", () => {
+      describe("Latest period", () => {
+        it("does not touch if shifted period do not overlap with original", () => {
+          const filter = new RelativeTimeFilterClause({
+            reference: "time",
+            period: TimeFilterPeriod.LATEST,
+            duration: Duration.fromJS("P3D")
```

Comment:
You have a difference by one day in the previous tests set, maybe for consistency sake, it should be `P1D` here? Or `P2D`, I'm not sure if I understand it correctly. If you need a test for an edge case, add it separately. Such one may be valuable for a fixed time filter as well.

Reply:
I don't follow. It's basically product of each `TimeFilterPeriod` and case for longer and case for shorter duration.

Reply:
I supposed this value is similar in some way to the pair of `2010-01-01` and `2010-01-02` used in the test for fixed time filter, that's why the result is the same.

Reply:
I'm reusing `shift` so I suppose I should reuse real range (relative or fixed). I will unify those cases.